Replicator/tests/ReplicatorWalrusTest.cc

```diff
@@ -79,11 +98,65 @@ class ReplicatorWalrusTest : public ReplicatorAPITest {
         }
     }
 
-};
+    enum AuthType {
+        kAuthNone,
+        kAuthBody,
+        kAuthHeader
+    };
 
+    void notWalrus(AuthType authType =kAuthNone) {
```

Comment:
This is pretty weak function naming

Reply:
Totally agreed. It's function is to make up the difference that, for non-walrus, we have to supply credentials, whereas in the walrus mode, database "scratch" is in the guest mode.
What about, "withAuthentication(AuthType)" ?

Reply:
Changed function name to withServerBackedSG()